sql/core/benchmarks/MiscBenchmark-results.txt

```diff
@@ -0,0 +1,132 @@
+generate explode array wholestage off         9658 / 9820          1.7         575.7       1.0X
+generate explode array wholestage on          6953 / 7005          2.4         414.4       1.4X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+generate explode map:                    Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+generate explode map wholestage off         13024 / 13061          1.3         776.3       1.0X
+generate explode map wholestage on          10300 / 10372          1.6         613.9       1.3X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+generate posexplode array:               Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+generate posexplode array wholestage off      9770 / 9910          1.7         582.3       1.0X
+generate posexplode array wholestage on       7033 / 7104          2.4         419.2       1.4X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+generate inline array:                   Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+generate inline array wholestage off          9260 / 9264          1.8         551.9       1.0X
+generate inline array wholestage on           7040 / 7668          2.4         419.6       1.3X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+generate big struct array:               Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+generate big struct array wholestage off       391 /  476          0.2        6511.5       1.0X
+generate big struct array wholestage on        353 /  372          0.2        5884.0       1.1X
+
+
+================================================================================================
+generate regular generator
+================================================================================================
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+generate stack:                          Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+generate stack wholestage off               17179 / 17719          1.0        1024.0       1.0X
+generate stack wholestage on                13674 / 14112          1.2         815.0       1.3X
```

Comment:
May be need deep dive, which was 15.5 X before.

Reply:
Performance declines because of this commit: [Disable generate codegen since it fails my workload ](https://github.com/apache/spark/commit/b5c5bd98ea5e8dbfebcf86c5459bdf765f5ceb53).

Reply:
Although this is not related to this refactorying, ping @rxin and @kiszk because @kiszk seemed to want to fix the root cause of the failure. 
> @rxin which operation in Generator makes failure of your workload? [commit comment](https://github.com/apache/spark/commit/b5c5bd98ea5e8dbfebcf86c5459bdf765f5ceb53#commitcomment-26392819).